dbt/tests/generic/test_row_values_match_after_join.sql

```diff
@@ -2,41 +2,85 @@
 -- a subset of the values in the joined table, i.e. if the class of PINA
 -- from tableA is '212' and the class of PINA from tableB is '211' and '212',
 -- then tableA matches (returns no rows).
+--
+-- Required parameters:
+--
+--   * external_model (str): The name of the model to join to.
+--   * external_column_name (str): The name of the column in `external_model`
+--         to join to.
+--   * join_condition (str): The ON (or USING) portion of a JOIN clause,
+--         represented as a string. Note that in the case where ON is used,
+--         columns in the base model should be formatted like `model.{column}`
+--         while columns in the external model should be formatted like
+--         `external_model.{column}`, e.g. `ON model.pin = external_model.parid`.
+--         This is not necessary in the case of USING, since USING does not
+--         refer to tablenames directly.
+--   * group_by (list of str): The columns from the base model to pass to the
+--         GROUP BY function used in the test query. Unlike `join_condition`,
+--         these column names do not have to be prefixed with `model.*`, since
+--         they are assumed to come from the base model for the test and not the
+--         external model.
+--
+-- Optional parameters:
+--
+--   * column_alias (str): An alias to use when selecting the column from the
+--         base model for output. An alias is required in this case because
+--         the column must be aggregated. Defaults to "model_col".
+--   * external_column_alias (str): An alias to use when selecting the column
+--         from the external model for output. Defaults to "external_model_col".
+--   * additional_select_columns (dict): Standard parameter for selecting
+--         additional columns from the base model for use in reporting failures.
+--         `model.{column_name}`, `external_model.{external_column_name`}`, and
+--         the columns specified in the `group_by` parameter are already selected
+--         by default and do not need to be included using this parameter.
+--         See the `format_additional_select_columns` macro for more information.
+--         Note that while this parameter can often be either a string or a dict,
+--         a dict is required for this test and it must contain the `alias`
+--         and `agg_func` parameters given that results are grouped by the
+--         columns specified in `group_by`.
+--
 {% test row_values_match_after_join(
     model,
-    column,
+    column_name,
     external_model,
-    external_column,
-    column_name="model_col",
-    external_column_name="external_model_col",
-    join_columns=[],
+    external_column_name,
+    join_condition,
```

Comment:
I considered using a data structure like a list of dicts to represent a more flexible set of join conditions, but ultimately it felt too restrictive for a test that I'm anticipating we'll want to use pretty widely in the future. Here's an example of the set of conditions that I had in mind while designing this:

```yaml
join_condition: |
  ON model.pin = external_model.parid
  AND model.year = external_model.taxyr
  AND external_model.rolltype != 'RR'
  AND external_model.procname = 'CCAOVALUE'
  AND external_model.deactivat IS NULL
  AND external_model.valclass IS NULL
```

It's possible we could use some combination of params and data structures to support joins like this, but allowing an arbitrary join condition seemed like the most flexible approach. I'm curious what you think!

Reply:
I agree. It's a little haphazard, but the string join condition is the most flexible. Let's keep it.